### PR TITLE
Update tags to allergens

### LINKED ISSUE #70: Update the current tag system to “allergen”

### PR DESCRIPTION
**Pull request by kinjalagarwal1810**
closes #70

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -385,10 +385,10 @@
     JAVADOC CHECKS
     -->
 
-    <!-- Checks that all block-tags are ordered correctly. -->
+    <!-- Checks that all block-allergens are ordered correctly. -->
     <module name="AtclauseOrder"/>
 
-    <!-- Checks that Javadoc block tags appear only at the beginning of the line. -->
+    <!-- Checks that Javadoc block allergens appear only at the beginning of the line. -->
     <module name="JavadocBlockTagLocation"/>
```
Comment (kokerinks) [inline config/checkstyle/checkstyle.xml:392]: not too sure whether the Tag in JavadocBlockTagLocation should be changed?

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -400,7 +400,7 @@
     <!-- Checks that each non-empty line in Javadoc has whitespace after leading asterisk. -->
     <module name="JavadocMissingWhitespaceAfterAsterisk"/>
 
-    <!-- Checks that for block tags, indentation of continuation lines is at least 4 spaces. -->
+    <!-- Checks that for block allergens, indentation of continuation lines is at least 4 spaces. -->
     <module name="JavadocTagContinuationIndentation"/>
```
Comment (kokerinks) [inline config/checkstyle/checkstyle.xml:405]: Same goes for here^ should the Tag in JavadocTagContinuationIndentation be changed ?

**Review by kokerinks — CHANGES_REQUESTED** [review]
Do review the 2 pointed out, im not too sure whether need to change also but if dn then can just ignore 👍 LGTM!

**Review by kokerinks — APPROVED** [review]

**Review by kab-dot — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/model/allergen/Allergen.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Allergen in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidAllergenName(String)}
+ */
+public class Allergen {
+
+    public static final String MESSAGE_CONSTRAINTS = "Allergens names should be alphanumeric";
```
[the remark this answers is not in the document]
Comment (billyhoce) [reply src/main/java/seedu/address/model/allergen/Allergen.java:12]: I think can just keep as alphanumeric, incase some people have certain medical conditions that include numbers like G6PD deficiency, and the user wants to add that as like an allergen tag for some reason!

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 4]
```diff
@@ -16,34 +17,32 @@
...
-                getTagSet("family"), new Points("0"), getOrderArrayList()),
+                getAllergenSet("PN"), new Points("0"), getOrderArrayList()),
             new Person(new Name("Irfan Ibrahim"), new Phone("92492021"), new Email("irfan@example.com"),
                 new Address("Blk 47 Tampines Street 20, #17-35"), new MembershipPoints(),
-                getTagSet("classmates"), new Points("0"), getOrderArrayList("Cheese Tart x 6@2023-06-30T23:23:23")),
+                getAllergenSet("V"), new Points("0"), getOrderArrayList("Cheese Tart x 6@2023-06-30T23:23:23")),
             new Person(new Name("Roy Balakrishnan"), new Phone("92624417"), new Email("royb@example.com"),
                 new Address("Blk 45 Aljunied Street 85, #11-31"), new MembershipPoints(),
-                getTagSet("colleagues"), new Points("0"),
+                getAllergenSet("TN"), new Points("0"),
```
Comment (billyhoce) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:45]: Sorry, but what are "LF", "TN", "EG", "PN", and "V"? HAHA

**Review by billyhoce — APPROVED** [review]
Small clarifications required, but code LGTM!

File: src/main/java/seedu/address/model/util/SampleDataUtil.java (hunk 4 above)
Comment (kinjalagarwal1810) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:45]: Lactose-free, Tree nuts, Egg, Peanuts, Vegan

**Pull request merged**